### PR TITLE
[IE VPU] Dynamic Concat fixes

### PR DESCRIPTION
**Pull request by Maxim-Doronin**
### Description
* Fixed the condition to select dynamic Concat in dynamic cases
* Added supporting int32 in Eltwise Divide

### Task
#-33110

### CODE REVIEW

**Comment by andrejsokolov** [comment]
@Maxim-Doronin, there are some failed tests.

**Comment by gladilov-gleb** [comment]
@ababushk @lazarevevgeny 

Could you please check if failures of 

TransformationTests.ConstFoldingPriorBoxClustered
TransformationTests.ConstFoldingPriorBoxClusteredSubgraph

are connected to merge of https://github.com/openvinotoolkit/openvino/pull/785 ?

**Comment by lazarevevgeny** [comment]
@itikhono , could you answer the question from @gladilov-gleb above?

**Comment by itikhono** [comment]
> @ababushk @lazarevevgeny
> 
> Could you please check if failures of
> 
> TransformationTests.ConstFoldingPriorBoxClustered
> TransformationTests.ConstFoldingPriorBoxClusteredSubgraph
> 
> are connected to merge of #785 ?

@gladilov-gleb @lazarevevgeny  Tests passed on ubuntu18 with the latest master and preсommit was successful in #785. Looks strange.

**Comment by Maxim-Doronin** [comment]
@andrejsokolov, please, review changes

File: inference-engine/src/vpu/graph_transformer/src/stages/concat.cpp [hunk 1]
```diff
@@ -293,10 +293,10 @@ void FrontEnd::parseConcat(
     // be replace with Data <-> Data edges.
     auto inferRequirement = ConcatInferRequirement::CanBeReplaced;
     if (auto concatOp = std::dynamic_pointer_cast<ngraph::opset3::Concat>(layer->getNode())) {
-        inferRequirement = concatOp->get_output_as_single_output_node(0)->get_type_info() ==
-                      ngraph::vpu::op::DynamicShapeResolver::type_info
-                      ? ConcatInferRequirement::NeedToInfer
-                      : ConcatInferRequirement::CanBeReplaced;
+        inferRequirement = concatOp->get_input_source_output(0).get_node_shared_ptr()->get_type_info() ==
```
Comment (gladilov-gleb) [inline inference-engine/src/vpu/graph_transformer/src/stages/concat.cpp:296]: @GlebKazantaev please take a look to check nGraph API usage

**Pull request merged**